src/api_client.rs

```diff
@@ -292,8 +287,22 @@ impl Client {
         unimplemented!("")
     }
 
-    pub async fn get_node_version() -> Result<String, Error> {
-        unimplemented!("")
+    pub async fn get_node_version(&self) -> String{
+        let result: serde_json::Value = serde_json::from_str(
+            &self
+                .http_get(&"eth/v1/node/version")
+                .await
+                .unwrap()
+                .text()
+                .await
+                .unwrap()
+        )
+        .unwrap();
+
+
+        return result["data"]["version"].to_string();
```

Comment:
this implementation does what you need in the happy case but if there are any failures during the call then the `unwrap`s will panic and cause the process to terminate. not nice UX and if in the event there happens to be value at stake (e.g. someone is using this in some validator software) then it would be a bad day.

another general note is to avoid duplicating code -- this helps with the maintenance burden (more distinct lines of code, more stuff to keep track of, debug etc.) and also helps w/ correctness -- this routine of "make an HTTP call and deserialize into one of the standard types in this library" is common across all of the methods in this module so it makes sense to define one common routine they can call share where possible.

if there is a bug in this routine then we can fix it in one place, not like 20 disparate implementations.

so I'd suggest using the `get` method something like this:
```rust
self.get(endpoint);
```

now we can look at the return type to see how to handle what we get back; and it turns out this method is generic over the success type `T`, specifically:

`Result<T, Error>`, where if we refer to the trait bounds on `T` we see it is `serde`-enabled.

so in this case we can direct the Rust compiler to generate the right code for us for our specific `T` (and `T` can be different at each call site which is where we get our polymorphism / ability to re-use this across all of the other functions)

so if we give a type hint:

```rust
let result: T = self.get(endpoint);
```

so what `T` do we want? for this we need to know a bit about the beacon APIs and the generally will return either a "success" response as JSON with a top-level object containing a `"data"` key OR an error that has a single object with the keys `"code"` and `"message"`.

this endpoint in particular https://ethereum.github.io/beacon-APIs/#/Node/getNodeVersion returns something like this:

```json
{
  "data": {
    "version": "Lighthouse/v0.1.5 (Linux x86_64)"
  }
}
```

OR

```json
{
  "code": 500,
  "message": "Internal server error"
}
```

so the "payload" of this success response is an object with the key `"version"` with the value of the version string we want to return to the caller.

from here, there is some nesting of types to handle the possible paths of success/error I just laid out:

to handle the first decision (success or error), we have the `ApiResult` type which is included in the `get` method. this is bc we will be using this pattern across the entire set of API methods so it makes sense to have one re-useable implementation. and the `serde` infra is set up to look at the response and try to figure out which case it is

so if we let `serde` handle this first part with the `ApiResult` type, we are left with this:

```json
{
    "version": "Lighthouse/v0.1.5 (Linux x86_64)"
}
```
so we then we need to specify a `T` that matches the rest of the success "shape" for this API method
and this is where you can use the `VersionData` data type you defined:

```rust
let result: VersionData = self.get(endpoint);
```

but if you do this you'll see the type still don't line up bc the return type is `Result<T, Error>` and while we specified the `T`, `T` is not equal to `Result<T, Error>`. we could write this out explicitly but rust has a handy `?` operator to handle this common pattern  so what we ultimately want is this:

```rust
let result: VersionData = self.get(endpoint)?;
```

and now we are left with one more step bc this function returns *just* the version string to the caller. there is no reason why the caller needs the extra wrapper as the information it communicates has already been conveyed in another way.

so then we can just return the inner data:

```rust
let data: VersionData = self.get(endpoint).await?;
Ok(data.version)
```

and note the `await` that had been left out until now and the fact that we need to wrap the return value in the right `Result` variant `Ok`.

if we do this, we can apply this pattern to most of the other methods here with a minimum of code and maximum of ease.

some endpoints will need custom types like `VersionData` as "adapters" from the beacon APIs and what makes sense for this library to expose. some endpoints (like the `events` category  will need entirely separate handling bc they do very different things)